File: frontend/src/types/anime/Anime.ts
export interface Anime {
  id: number
  title_romaji: string
  title_english: string
  native: string
  description: string
  coverImage: string
  bannerImage: string
  cover_color_hex: string
  studio_id: number
  episode_count: number
  episode_duration: number
  average_score: number
  popularity: number
  start_date: string
  end_date: string
  seasonYear: number
  format: string
  status: string
  source: string
  season: string
  genres: Genre[]
  studios: Studio[]
  characters?: Character[]
  externalLinks: ExternalLink[]
  trailer: {
    youtubeId: string
  }
  relations: RelatedMedia[]
}

export interface Genre {
  id: number
  name: string
}

export interface Studio {
  id: number
  name: string
  isMain: boolean
}

export interface AnimeCharacter {
  characters: Character[]
}

export interface Character {
  id: number
  name_full: string
  name_native: string
  age?: string
  description: string
  dateOfBirth?: string
  image: string
  role: string
  languageV2: string
  staff: Staff[]
}

export interface Staff {
  id: number;
  name_full: string;
  name_native: string;
  languageV2: string;
  image: string;
  description: string;
  primaryOccupations: string[];
  gender: string;
  dateOfBirth?: string;
  dateOfDeath: string;
  age?: number;
  yearsActive: number[];
  homeTown: string;
  created_at: string;
  updated_at: string;
  notableWorks?: {
    title: string;
    role: string;
  }[];
}

export interface ExternalLink {
  id: string;
  url: string;
  site: string;
}

export interface RelatedMedia {
  id: number
  title: string
  image: string
  relationType: string
}

File: backend/schemas/v1/anime.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date
from schemas.v1.shared.genre import GenreSchema  # Importa el schema de Genre
from schemas.v1.studio import StudioSchema  # Importa el schema de Studio
from schemas.v1.season import SeasonSchema  # Importa el schema de Season

class AnimeBase(BaseModel):
    id: int
    title_romaji: str
    title_english: Optional[str] = None
    native: Optional[str] = None
    description: Optional[str] = None
    coverImage: Optional[str] = None
    bannerImage: Optional[str] = None
    episode_count: Optional[int] = None
    episode_duration: Optional[int] = None
    average_score: Optional[float] = 0
    popularity: Optional[int] = 0
    season_id: Optional[int] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    seasonYear: Optional[int] = None
    format: Optional[str] = None
    status: Optional[str] = None
    source: Optional[str] = None

class AnimeCreate(AnimeBase):
    pass

class AnimeUpdate(AnimeBase):
    pass

class Anime(AnimeBase):
    season: Optional[str] 
    genres: List[GenreSchema]
    studios: List[StudioSchema]


    class Config:
        from_attributes = True